ARCHITECTURE.md

```diff
@@ -0,0 +1,97 @@
+
+The interface to the cache is described in the package mds-cache.  
+# Messaging
+Although the current services in MDS-core do not communicate with each other, LADOT is using some data-consumers that process Agency-ingested events and telemetry via a publish-and-subscribe message bus.
+
+Services in a service-oriented-architecture can communicate with each other via the same types of interfaces used by the outside clients (e.g. REST or SOAP), or they can use an internal message bus. We chose the latter for perceived wins in speed and efficiency. After considering knative eventing and kafka, we settled on [NATS](https://nats.io) as a more stable and low-operational-overhead solution. We have attempted to abstract away the message interface, however, such that we could later decide to plug in a different message bus.
+
+NATS is recognized as an incubating [CNCF](https://www.cncf.io) [open source project](https://github.com/nats-io) made available under the [Apache License](https://www.apache.org/licenses/LICENSE-2.0.html) and supporting a number of contemporary software languages.
+# Testing & Validation
+The repo currently has integration tests with code-coverage minimums on a per-project basis.  [TODO add more]
+# Versioning
+The services in mds-core will follow the MDS spec’s requirements for [API versioning](https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/general-information.md).
+# Shared Libraries
+The mds-core repo contains a number of libraries, including the previously mentioned mds-db (postgres wrapper) and mds-cache (redis wrapper).
+## mds-api-authorizer
+Code to extract standard and custom claims from Basic or Bearer Authorization headers 
+## mds-api-helpers
+Simple utilities for construction APIs. For example, processing paging parameters or formating JSON API links
+## mds-api-server
+Standard API server which implements middleware and endpoints that are common to all MDS APIs.
+## mds-logger
+Basic logging wrapper that includes integration with Slack and Pushover as well as standard console-output.
+
+## mds-orm
+Code for managing connections to Postgres and schema migrations.
+## mds-providers
+Collection of information about supported micro-mobility providers.
+## mds-stream
+Collection of methods to send messages to NATS-Streaming. Currently supports sending vehicle events and telemetry.
+## mds-types
+Collection of TypeScript descriptions of types that are shared among components, including Agency’s notion of Device and VehicleEvent, event type enums for Provider and Agency, etc.
+## mds-utils
+Grab-bag of shared utility methods. Including, but not limited to: Programmatic representation of the MDS-Agency State Machine, date-time converters, and geospatial analysis methods.
+## mds-web-sockets
+Implementation of a basic websocket model for MDS. Enables real-time external consumption of data through a pub/sub model. See package documentation for specific examples. 
```

Comment:
Should we mention that it's probably a better practice to use NATS instead of hooking into the web sockets directly?